spire-flex/templates/agent-configmap.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $.Release.Name }}-agent-config
+  namespace: {{ $.Release.Namespace }}
+data:
+  agent_config: |
+    agent {
+    }
+    
+    plugins {
+    }
+    {{ if ne (((.Values.agent).healthCheck).enable) false }}
+    health_checks {
+      listener_enabled = true
+      bind_address = {{ default "localhost" (((.Values.agent).healthCheck).bindAddress) | quote }}
+      bind_port = {{ default 80 (((.Values.agent).healthCheck).bindPort) }}
+      live_path = {{ default "/live" (((.Values.agent).healthCheck).livePath) | quote }}
+      ready_path = {{ default "/ready" (((.Values.agent).healthCheck).readyPath) | quote }}
```

Comment:
There a reason not to put the defaults in the chart's values.yaml file? It makes it more discoverable that way.

Reply:
If it is in the chart's value.yaml file, the value will always be set, which means it can not be unit tested for when it is missing.
We need to trust in our documentation as the main way to discover the values.  That's why I put the documentation submission in along with the code submission.

Reply:
Actually, the quality of charts out there is so bad that nobody has default values, they just configure every value every time in the value.yaml files.
If you think me wrong, they why would Helm even have a "default" function?  The value would already always be set.

Reply:
https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function and scroll down to:

> In an actual chart, all static default values should live in the values.yaml, and should not be repeated using the default command (otherwise they would be redundant)

Reply:
> https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function and scroll down to:
> 
> > In an actual chart, all static default values should live in the values.yaml, and should not be repeated using the default command (otherwise they would be redundant)

This is very interesting, and it directly contradicts a few items that good user interfaces provide (don't make me think, designing highly usable user interfaces, et. al.)  I'm not going to take a stand here, but I would like to attempt a different approach.  If that approach fails, we will implement the appraoch suggested above.

I think they're just stating that one won't have defaults in the charts to update, which is a kind of developer convenience, but being presented with too much information that is mostly defaults is a different kind of user interface sin.  

While some love it and some hate it, Apache's Maven is an example of "the defaults make sense and aren't visible.  Only the deviations from the norm or items important enough to a team that they are explicitly added are visible should be obvious".  I know that a lot of people coming from Makefiles completely hated Maven, but the approach has some benefits, namely that you don't miss the changes in a sea of defaults.  

In any case, now I don't want to was too philosophical and go too far off the rails on this one.  If we have time to burn in the meeting, I'd like to cover this at a lower priority, unless it's a key point you feel strongly about.